stablehlo/reference/Interpreter.h

```diff
@@ -32,6 +33,14 @@ namespace stablehlo {
 llvm::Expected<llvm::SmallVector<Tensor>> eval(func::FuncOp func,
                                                llvm::ArrayRef<Tensor> args);
 
+/// Evaluates an mlir::Region `region` using the runtime values `runtimeArgs`
+/// corresponding to the arguments of the containing block, assuming that the
+/// region has only one block. Interprets the operations within the block and
+/// returns the runtime values for the terminator's arguments.
+llvm::Expected<llvm::SmallVector<Tensor>> eval(
+    Region &region, llvm::ArrayRef<Tensor> runtimeArgs,
+    const InterpreterScope *const parentScope);
```

Comment:
Eve though we are not allowing copy of `InterpreterScope` objects, we are passing the parent Scope to function `evalRegion` via pointer.
 

It would be ideal if `evalRegion` can take the parent pointer as `const InterpreterScope &parent`. But we could not use that because 
1. `evalRegion` is called either by `evalFunc` or `eval<Op>` and `evalFunc` needs to pass some value denoting "no scope". Since "optional reference" is not a thing, the choice I made is `nullptr`.
The `const InterpreterScope * const` signature is to mimic the affect of a const reference to disallow mutation of the parent pointer.